packages/squiggle-lang/src/rescript/Reducer/Reducer_MathJs/Reducer_MathJs_ToExpression.res

```diff
@@ -0,0 +1,86 @@
+module ErrorValue = Reducer_ErrorValue
+module ExpressionValue = ReducerInterface.ExpressionValue
+module ExtressionT = Reducer_Expression_T
+module JavaScript = Reducer_Js
+module Parse = Reducer_MathJs_Parse
+module Result = Belt.Result
+
+type expression = ExtressionT.expression
+type expressionValue = ExpressionValue.expressionValue
+type errorValue = ErrorValue.errorValue
+
+let rec fromNode = (mjnode: Parse.node): result<expression, errorValue> =>
+  Parse.castNodeType(mjnode)->Result.flatMap(typedMjNode => {
+    let fromNodeList = (nodeList: list<Parse.node>): result<list<expression>, 'e> =>
+      Belt.List.reduceReverse(nodeList, Ok(list{}), (racc, currNode) =>
+        racc->Result.flatMap(acc =>
+          fromNode(currNode)->Result.map(currCode => list{currCode, ...acc})
+        )
+      )
+
+    let caseFunctionNode = fNode => {
```

Comment:
why is this called ``caseFunctionNode``? Did you mean ``"cast"FunctionNode``, by chance?

Reply:
Fixed.
Problem. Do a mass renaming. Run unit tests. Passing. OK.  In fact not OK, because sometimes there are renaming artifacts that managed to pass the tests. I am not able to read through the code every time. So far executed tests 153 times for renamings.

Reply:
RESOLVED